here-naksha-app-service/src/test/java/com/here/naksha/app/service/CollectionModificationTest.java

```diff
@@ -0,0 +1,108 @@
+        assertThat(updateSpaceResp1).hasStatus(200);
+
+        // When: New feature is created
+        HttpResponse<String> createResp = nakshaClient.post("hub/spaces/" + REGULAR_SPACE_ID + "/features", createFeatureJson, streamId);
+        assertThat(createResp).hasStatus(200);
+
+        // And: This feature is updated
+        HttpResponse<String> updateResp = nakshaClient.put("hub/spaces/" + REGULAR_SPACE_ID + "/features/" + featureId, updateFeatureJson,
+                streamId);
+        assertThat(updateResp).hasStatus(200);
+
+        // And: This feature is deleted
+        HttpResponse<String> deleteResp = nakshaClient.delete("hub/spaces/" + REGULAR_SPACE_ID + "/features/" + featureId, streamId);
+        assertThat(deleteResp).hasStatus(200);
+
+        // And: Client queries activity log space for this feature
+        String featureIdNamespaceQuery = urlEncoded("p.@ns:com:here:xyz:log.id") + "=" + featureId;
+        HttpResponse<String> getResp = nakshaClient.get("hub/spaces/" + ACTIVITY_SPACE_ID + "/search?" + featureIdNamespaceQuery, streamId);
+
+        // And: Expected ActivityLog response matches the response
+        assertThat(getResp)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedActivityResp)
+        ;
+
+        // When: Regular space is updated to disable activity history and enable autoPurge
+        HttpResponse<String> updateSpaceResp2 = nakshaClient.put("hub/spaces/"+REGULAR_SPACE_ID, updateSpaceJson2, streamId);
+        assertThat(updateSpaceResp2).hasStatus(200);
+
+        // Then:
+        HttpResponse<String> createResp2 = nakshaClient.post("hub/spaces/" + REGULAR_SPACE_ID + "/features", createFeatureJson2, streamId);
+        assertThat(createResp2).hasStatus(200);
+
+        // And: This feature is updated
+        HttpResponse<String> updateResp2 = nakshaClient.put("hub/spaces/" + REGULAR_SPACE_ID + "/features/" + featureId2, updateFeatureJson2,
+                streamId);
+        assertThat(updateResp2).hasStatus(200);
+
+        // And: This feature is deleted
+        HttpResponse<String> deleteResp2 = nakshaClient.delete("hub/spaces/" + REGULAR_SPACE_ID + "/features/" + featureId2, streamId);
+        assertThat(deleteResp2).hasStatus(200);
+
+        // And: Client queries activity log space for this feature
+        String featureIdNamespaceQuery2 = urlEncoded("p.@ns:com:here:xyz:log.id") + "=" + featureId2;
+        HttpResponse<String> getResp2 = nakshaClient.get("hub/spaces/" + ACTIVITY_SPACE_ID + "/search?" + featureIdNamespaceQuery2, streamId);
+
+        // And: Expected ActivityLog response matches the response
+        assertThat(getResp2)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedActivityResp2)
+        ;
+    }
+
+
+
+
+    private static String urlEncoded(String text) {
```

Comment:
urlEncoded is already part of TestUtil. Let's use the same and this function can be removed.

Reply:
oh thanks, somehow my IDE couldn't find this from TestUtil, have to manually write the import line.